plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -231,7 +242,7 @@ private static Object resolve(String pathElement, String fullPath, Object contex
      * @throws IllegalArgumentException if the path is null, empty, invalid or if the field doesn't exist.
      */
     public void appendFieldValue(String path, Object value) {
-        setFieldValue(path, value, true);
+        setFieldValue(TemplateService.wrap(path), ValueSource.wrap(value), true);
```

Comment:
does this mean that a path provided as string doesn't support a template? I guess so and that way the processors will choose whether they support templating or not rather than supporting it at the IngestDocument level? I wonder if we need this wrapping though for simple strings. Given that a template in the end returns a string, do we really need to wrap a string into the StringTemplate? seems like strings could just skip the execution part as a whole?